### ISSUE TITLE
Plugin conflicting with esformatter-semicolons

### ISSUE THREAD

**Issue by sircharleswatson** [twolfson/esformatter-var-each#2]
I'm having issues where this plugin is conflicting with esformatter-semicolons and producing errors that cause my app to crash.

you can see more details in millermedeiros/esformatter#326

I can get it working if I run esformatter with just the semicolon plugin first and then run the rest, but I'd like to not have to do that.

**Comment by twolfson** [comment]
As mentioned in millermedeiros/esformatter#326, the conflicting test case can be found here:

https://gist.github.com/twolfson/0d76da04b19870c6ab1f/60c8fa91487ac75b4d365a054deedb506a392075

**Comment by twolfson** [comment]
I began digging into this but then began a refactor due to a lot of code which should be available in other modules (e.g. tokens). I believe the issue lies in us conditionally adding a line break but am not sure yet. I was hoping a refactor would sanely correct this but I wound up draining too much time.

I will take another look at this by the end of next weekend.

**Comment by sircharleswatson** [comment]
@twolfson Thanks a ton! I really appreciate it. I look forward to this issue being resolved :smile:

**Comment by twolfson** [comment]
Alright, took another shot at it. After about 3/4 times of getting frustrated, it's rewritten and better commented (hopefully). Your use case should be fixed when you upgrade:

```bash
npm install esformatter-var-each@latest
```

Thanks for the bug report =)

**Comment by sircharleswatson** [comment]
@twolfson I just tried running this with the latest and I get `Cannot read property 'indexOf' of undefined`

I removed your package and the error went away. :/

**Comment by twolfson** [comment]
Damn, I guess the refactor must have broken something. Fwiw, we added a regression test for the past error so that definitely can't be the issue:

https://github.com/twolfson/esformatter-var-each/blob/2.0.1/test/test-files/intermediate-indented-var-each-semicolon-less.js

The `indexOf` is more likely to be from the AST manipulation code than the code itself. I will take a look at reproducing with the previously provided example on `esformatter` later on this week.

**Comment by sircharleswatson** [comment]
I actually tried the previously provided sample code and it worked fine. :/

**Comment by twolfson** [comment]
Oh, damn =(

**Comment by sircharleswatson** [comment]
I manually fixed the var-each stuff in this file so I'll try and run it with it already fixed and see if i still get the same error.

**Comment by twolfson** [comment]
If the code is closed, then I guess the best we can do is stare at the stack trace. Can you copy/paste the full error you are getting from `esformatter`?

**Comment by sircharleswatson** [comment]
is there a way to get a verbose output from esformatter? if not, literally all that it outputs is: `Cannot read property 'indexOf' of undefined`

**Comment by twolfson** [comment]
The programmatic usage might get you a better stack trace:

```js
// Load in our dependencies
var fs = require('fs');
var esformatter = require('esformatter');

// Load in our configuration
var config = require('./config.json');

// Load in and transform our file
var input = fs.readFileSync('path/to/js.js', 'utf8');
var output = esformatter.format(input, config);

// Output our transformed JS
fs.writeFileSync('transformed.js', output, 'utf8');
```

**Comment by aheckmann** [comment]
I'm seeing this error as well. 

```js
TypeError: Cannot read property 'indexOf' of undefined
    at exports._transformNode (... /esformatter-var-each/lib/esformatter-var-each.js:197:68)
```

It looks like this is caused when a `var` is declared within a switch case statement because the `var` declarations `parent` node (a SwitchCase) has no `body` property.

```js
switch (testing) {
  case 'match':
    var fails = 1337;
}
```

**Comment by aheckmann** [comment]
I could open a new issue since the source of the bug is independent of any other plugins.

**Comment by twolfson** [comment]
@aheckmann I have successfully reproduced the issue. Thanks for reporting it, it's great to finally have a test case.

I will attempt to get this resolved within the next week. Feel free to open a new issue, although don't consider it necessary.

**Comment by twolfson** [comment]
Alright, we have patched up the issue in `2.1.0`.

Ironically, I have already patched this issue in `jscs` and patched it there. However, I guess I forgot about the edge case here.

https://github.com/jscs-dev/node-jscs/pull/1243

Thanks @aheckmann for the test case =)

**Comment by aheckmann** [comment]
:+1:
